calico_node/filesystem/startup.py

```diff
@@ -261,7 +261,7 @@ def main():
     # This is required because the confd templates expect to be able to fill in
     #  the IP by fetching it from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("IP=" + str(ip))
```

Comment:
I think that doing it at line 250 will  only bandaid the fact that we now receive `IPAddress` objects from `get_host_ips`, but all our calico-containers functions expect strings to be passed around. We should either be getting a string from `get_host_ips`, or all the below functions should be updated to take `IPAddress` objects

I could use some input on that, @robbrockbank @TrimBiggs ?